### PR TITLE
[excel] (Tables) Add an article about working with JSON

### PR DESCRIPTION
**Pull request by AlexJerabek**
JSON provides a convenient way to pass data to and from scripts. It's also the default mechanism for many web services and Power Automate connectors. This article seeks to contextualize the topics within the Office Scripts world.

### CODE REVIEW

**Review by davidchesnut — CHANGES_REQUESTED** [review]
Nice. Had a few suggestions...

File: docs/develop/use-json.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+JSON data in Power Automate can be passed as a generic object array. Add an `object[]` property to the script to accept that data.
+
+```typescript
+// For Power Automate, replace the main signature in the previous sample with this one
+// and remove the sample data.
+function main(workbook: ExcelScript.Workbook, jsonData: object[]) {
+```
+
+You'll then see an option in the Power Automate connector to add `jsonData` to the **Run script** action.
```
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply docs/develop/use-json.md:125]: That's what the next screenshot shows. Power Automate flows are assembled in their own site/UI.

**Review by davidchesnut — APPROVED** [review]
Looks good!

File: docs/develop/use-json.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+> ```
+
+### Get JSON data from external sources
+
+There are two ways to import JSON data into your workbook through an Office Script.
+
+- As a [parameter](power-automate-integration.md#main-parameters-pass-data-to-a-script) with a Power Automate flow.
+- With a `fetch` call to an [external web service](external-calls.md).
+
+#### Modifying the sample to work with Power Automate
```
Comment (lindalu-MSFT) [inline docs/develop/use-json.md:119]:
```suggestion
#### Modify the sample to work with Power Automate
```

File: docs/develop/use-json.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+// For Power Automate, replace the main signature in the previous sample with this one
+// and remove the sample data.
+function main(workbook: ExcelScript.Workbook, jsonData: object[]) {
+```
+
+You'll then see an option in the Power Automate connector to add `jsonData` to the **Run script** action.
+
+:::image type="content" source="../images/json-parameter-power-automate.png" alt-text="An Excel Online (Business) connector showing a Run script action with the jsonData parameter.":::
+
+#### Modifying the sample to use a `fetch` call
```
Comment (lindalu-MSFT) [inline docs/develop/use-json.md:133]:
```suggestion
#### Modify the sample to use a `fetch` call
```

File: docs/develop/use-json.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+  console.log(jsonArray);
+}
+
+```
+
+## See also
+
+- [External API call support in Office Scripts](external-calls.md)
+- [Sample: Use external fetch calls in Office Scripts](../resources/samples/external-fetch-calls.md)
+- [Run Office Scripts with Power Automate](power-automate-integration.md)
```
Comment (lindalu-MSFT) [inline docs/develop/use-json.md:243]:
```suggestion
- [Run Office Scripts with Power Automate](power-automate-integration.md)
- [Office Scripts Dev Center](https://developer.microsoft.com/office-scripts)
```

File: docs/resources/samples/external-fetch-calls.md [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
 ---
 
 # Use external fetch calls in Office Scripts
 
-This script gets basic information about a user's GitHub repositories. It shows how to use `fetch` in a simple scenario. For more information about using `fetch` or other external calls, read [External API call support in Office Scripts](../../develop/external-calls.md)
+This script gets basic information about a user's GitHub repositories. It shows how to use `fetch` in a simple scenario. For more information about using `fetch` or other external calls, read [External API call support in Office Scripts](../../develop/external-calls.md). For information about working with [JSON]](https://www.w3schools.com/whatis/whatis_json.asp) objects, like what is returned by the GitHub APIs, read [Use JSON to pass data to and from Office Scripts](../../develop/use-json.md).
 
 You can learn more about the GItHub APIs being used in the [GitHub API reference](https://docs.github.com/rest/reference/repos#list-repositories-for-a-user). You can also see the raw API call output by visiting `https://api.github.com/users/{USERNAME}/repos` in a web browser (be sure to replace the {USERNAME} placeholder with your GitHub ID).
```
Comment (lindalu-MSFT) [inline docs/resources/samples/external-fetch-calls.md:12]:
```suggestion
Learn more about the GItHub APIs being used in the [GitHub API reference](https://docs.github.com/rest/reference/repos#list-repositories-for-a-user). You can also see the raw API call output by visiting `https://api.github.com/users/{USERNAME}/repos` in a web browser (be sure to replace the {USERNAME} placeholder with your GitHub ID).
```
The two "you cans" in a row jumped out at me!

Comment (AlexJerabek) [reply docs/resources/samples/external-fetch-calls.md:12]: If the Yucatan toucan can use two "you cans", you can too!

File: docs/develop/use-json.md [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+
+This article focuses on using JSON with Office Scripts. We recommend you first learn more about the format from articles such [JSON Introduction](https://www.w3schools.com/js/js_json_intro.asp) from W3 Schools.
+
+## Parse JSON data into a range or table
+
+Arrays of JSON objects provide a consistent way to pass rows of table data between applications and web services. In these cases, each JSON object represents a row, while the properties represent the columns. An Office Script can loop over an JSON array and reassemble it as a 2D array. This array is then set as the values of a range and stored in a workbook. The property names can also be added as headers to create a table.
+
+The following script shows JSON data being converted to a table. Note that the data is not taken from an external source. That is covered later in this article.
```
Comment (jeremy-msft) [inline docs/develop/use-json.md:18]:
```suggestion
The following script shows JSON data being converted to a table. Note that the data is not taken from an external source; that is covered later in this article.
```

File: docs/develop/use-json.md [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+  { "Action": "Paste", "N": 1171, "Percent": 6.2 },
+  { "Action": "Clear", "N": 599, "Percent": 3.17 },
+  { "Action": "Insert", "N": 352, "Percent": 1.86 },
+  { "Action": "Delete", "N": 350, "Percent": 1.85 },
+  { "Action": "Refresh", "N": 314, "Percent": 1.66 },
+  { "Action": "Fill", "N": 286, "Percent": 1.51 },
+];
+
+/**
+ * This script converts JSON data to a Excel table.
```
Comment (jeremy-msft) [inline docs/develop/use-json.md:40]:
```suggestion
 * This script converts JSON data to an Excel table.
```

**Pull request merged**